docs: split README into one for each action (#46)
canonical/craft-actions: snapcraft/pack/README.md

# snapcraft/pack

Set up Snapcraft and pack a snap.

## Usage

```yaml
- name: Package snap with Snapcraft
  uses: canonical/craft-actions/snapcraft/pack@main
```

This installs and configures LXD and Snapcraft, then packs the snap

> at the root of the repository.

Author: Agreed on the examples, thanks for making an issue for that :)

### Inputs

> All inputs and their defaults.

Reviewer: I'm fine with "block" :+1:

> All inputs and their defaults.

Reviewer: Revising my original suggestion after coming back to this with fresh eyes.

```suggestion
The action can be catered to your project with the following inputs:
```
 This builds on [my last suggestion](https://github.com/canonical/craft-actions/pull/46/changes#r2615783958) and the more explicit mention of the defaults we added. Probably a little too much passive voice, but I think it's acceptable given the reference-ish doc we have here.

> All inputs and their defaults.

Reviewer: How about:

> The action can be catered to your project with the following inputs. Where applicable, inputs are listed with their default values.

```yaml
- name: Package snap with Snapcraft
  uses: canonical/craft-actions/snapcraft/pack@main
  with:
    # The channel to install the Snapcraft snap from.

>     channel: 'latest/stable'

Reviewer: ```suggestion
    # The channel to install the Snapcraft snap from.
    # Defaults to 'latest/stable'.
    channel: 'latest/stable'
```
Given the confusion you noticed around the defaults, I don't think it hurts to be explicit here even if it's slightly redundant.

    
    # The revision of the Snapcraft snap to install.
    # Overrides the `channel` option.
    revision: ''

    # The channel to install the LXD snap from.
    # If left blank, defaults to the current recommended channel for Snapcraft.

>     lxd-channel: ''

Reviewer: > If left blank

Is the behavior the same if it isn't set at all?

    
    # The location in your repository to run Snapcraft when packing. Defaults to the repository root.
    # 
    # In most simple cases, `snapcraft.yaml` is stored in the `snap/` directory at the repository root.
    # 
    # In repositories with more complex file hierarchies, the path could be elsewhere. For example, if your project
    # stores multiple build tool manifests in a `dist/` directory, and `snapcraft.yaml` is stored in `dist/snap/`,

>     # you would want to run Snapcraft at `dist/` instead of the repository root.

Reviewer: ```suggestion
    # you'd run Snapcraft at `dist/` instead of the repository root.
```
Small change for concision

    path: '.'
    

>     # Sets the build verbosity level for Snapcraft. Must be one of: "quiet", "brief", "verbose", "debug", or "trace" (the default).

Reviewer: ```suggestion
    # The build verbosity level for Snapcraft. Must be one of: 'quiet', 'brief', 'verbose', 'debug', or 'trace'.
    # Defaults to 'trace'.
```

# snapcraft/pack

When invoked, this action installs and configures LXD and Snapcraft before packing the snap
at the root of the repository.

## Usage

The snapcraft/pack action can be added to your repository's workflows with:

```yaml
- name: Package snap with Snapcraft
  uses: canonical/craft-actions/snapcraft/pack@main
```

### Inputs

> The action can be catered to your project with the following inputs. Where applicable, inputs are listed with their default values.

Author: hmm. Those "# Defaults to" comments were added later, they probably help the situation more than I'm giving them credit. I say let's try to just use those, and only add the "no default" comments if users seem lost.

# snapcraft/pack

When invoked, this action installs and configures LXD and Snapcraft before packing the snap
at the root of the repository.

## Usage

```yaml
- name: Package snap with Snapcraft
  uses: canonical/craft-actions/snapcraft/pack@main
```

This installs and configures LXD and Snapcraft, then packs the snap

> at the root of the repository.

Reviewer: Rather than reiterating the first sentence, how do you feel about changing this to something along the lines of:

````suggestion
## Usage

The snapcraft/pack action can be added to your repository's workflows with:

```yaml
- name: Package snap with Snapcraft
  uses: canonical/craft-actions/snapcraft/pack@main
```
````
